packages/react-native/ReactAndroid/src/test/java/com/facebook/react/fabric/events/TouchEventDispatchTest.kt

```diff
@@ -0,0 +1,681 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.react.fabric.events
+
+import android.util.DisplayMetrics
+import android.view.MotionEvent
+import com.facebook.react.bridge.*
+import com.facebook.react.fabric.FabricUIManager
+import com.facebook.react.uimanager.DisplayMetricsHolder
+import com.facebook.react.uimanager.events.TouchEvent
+import com.facebook.react.uimanager.events.TouchEventCoalescingKeyHelper
+import com.facebook.react.uimanager.events.TouchEventType
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentCaptor
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.times
+import org.powermock.api.mockito.PowerMockito.mockStatic
+import org.powermock.api.mockito.PowerMockito.`when` as whenever
+import org.powermock.core.classloader.annotations.PowerMockIgnore
+import org.powermock.core.classloader.annotations.PrepareForTest
+import org.powermock.core.classloader.annotations.SuppressStaticInitializationFor
+import org.powermock.modules.junit4.rule.PowerMockRule
+import org.robolectric.RobolectricTestRunner
+
+@PrepareForTest(Arguments::class, FabricUIManager::class)
+@SuppressStaticInitializationFor("com.facebook.react.fabric.FabricUIManager")
+@RunWith(RobolectricTestRunner::class)
+@PowerMockIgnore("org.mockito.*", "org.robolectric.*", "androidx.*", "android.*")
+class TouchEventDispatchTest {
+  @get:Rule var rule = PowerMockRule()
+  private val touchEventCoalescingKeyHelper = TouchEventCoalescingKeyHelper()
+
+  /** Events (1 pointer): START -> MOVE -> MOVE -> UP */
+  private val startMoveEndSequence =
+    listOf(
+      createTouchEvent(
```

Comment:
For all the `createTouchEvent`, `buildGestureEvent`, `buildGesture` and other private methods, could you use named parameters? It will make the tests easier to read.